### DISCUSSION TITLE
Fungsi Separate error/Kolom Baru

### DISCUSSION THREAD

**Discussion by pram9113** [teamalgoritma/community#778] [Q&A] [answered]
Halo Selamat Siang,

Saya mau buat kolom baru sesuai table di bawah ini, 
![image](https://github.com/teamalgoritma/community/assets/26856093/3d97894a-ae4b-4cb1-968a-650c26e6ad08)

Code yang saya coba 

`separate(df, col=Vehicle.Location, into=c('long', 'lat'), sep='-')`

Namu ada error terkait penggunaan fungsi sebagai 'chr' saya ubah ke Num dan Int hasilnya NA. Apakah dapat menggunakan 'mutate' 

Terima Kasih ya,

Pramadia Lazuardi

**Comment by ramdhinafinita** [comment]
Dear @pram9113 

Pertanyaan Anda telah kami terima dan sedang kami usahakan terkait solusinya. Mohon kesediaannya untuk menunggu, terima kasih.

Best regards,
Team Algoritma

**Comment by ramdhinafinita** [comment, marked as answer]
Dear @pram9113 

Berdasarkan sreenshot yang anda tampilkan terlihat data location yang seharusnya geometri menjadi character, jika memungkinkan unuk tidak perlu di pisah juga bisa, bisa di read menggunakan library sf atau sp. 
Namun, jika memang sulit. Berikut saran dari kami bisa menggunakan gsub:

```
# contoh df
df <- data.frame(location = c("POINT (-122.34301 47.659185)", "POINT (-122.20578 47.762405)"))

# Ekstrak longitude dan latitude menggunakan regex gsub
df$long <- as.numeric(gsub("POINT \\((-?\\d+\\.\\d+) (-?\\d+\\.\\d+)\\)", "\\1", df$location))
df$lat <- as.numeric(gsub("POINT \\((-?\\d+\\.\\d+) (-?\\d+\\.\\d+)\\)", "\\2", df$location))

# cek hasil
df
```
Bisa  dicoba terlebih dahulu, ini menggunakan operasi regex, sehingga seperti mengambil informasi dalam baris di kolom tertentu. 
Referensi:
https://www.programmingr.com/tutorial/gsub-in-r/
https://www.endmemo.com/r/gsub.php

**Reply by pram9113** [reply]
Dear @ramdhinafinita 

Terima kasih, solved!
